aom/src/main/java/com/nedap/archie/aom/CAttribute.java

```diff
@@ -372,7 +372,7 @@ public boolean isProhibited() {
     @Override
     @JsonIgnore
     public boolean isLeaf() {
-        return children != null && children.size() > 0;
+        return children == null || children.isEmpty();
```

Comment:
It look like `children` can't be null. `setChildren(null)` will make it an empty list.

Reply:
Correct. However, I'd like to keep this null check, just to be sure and to prevent problems after future code changes. We don't have compile time Kotlin style null checks, and these methods should always just work.